### PR TITLE
Simple guide to using docker and docker-compose with drand

### PR DESCRIPTION
**Pull request by CluEleSsUK**

### CODE REVIEW

**Comment by CluEleSsUK** [comment]
I think this is ready to rock

File: docker/README.md [hunk 1]
```diff
@@ -1,6 +1,70 @@
...
+## Prerequisites
+- install docker
+- install docker compose (probably bundled with docker these days)
+- run on linux :) (other platforms may work but YMMV)
+
+## Creation of a keypair
+
+Pull the latest drand image:
+```shell
+docker pull drandorg/go-drand:v1.5.3
```
Comment (AnomalRoil) [inline docker/README.md:12]: Does this need to be updated to point to the new github registry?

File: docker/README.md [hunk 2]
```diff
@@ -1,6 +1,70 @@
...
+
+Create a volume where you're going to store your keypairs and other config data
+```shell
+docker volume create drand
+```
+
+Next we must create a keypair and store it in the docker volume we've just created.
+
+```shell
+docker run --rm --volume drand:/data/drand drandorg/go-drand:v1.5.3 generate-keypair  --folder /data/drand/.drand --tls-disable --id default 0.0.0.0:8080
```
Comment (AnomalRoil) [inline docker/README.md:23]: same

File: docker/README.md [hunk 3]
```diff
@@ -1,6 +1,70 @@
...
+You should replace `0.0.0.0:8080` with your public IP address, e.g. `pl1-rpc.drand.sh:443`, as this key is how other nodes in the network
+will verify that they're talking to your node. 
+
+_Note_: access to this path should be firewalled to only allow connections from nodes in the relevant allowlist ([mainnet allowlist](https://github.com/drand/loe-mainnet-allowlist/) and [testnet allowlist](https://github.com/drand/loe-testnet-allowlist/))
+
+
+## Starting drand
+Finally we can start the docker container by running:
+```shell
+docker run --rm -d -p"8080:8080" -p"8888:8888" --name drand  --volume drand:/data/drand drandorg/go-drand:v1.5.3 start --tls-disable --private-listen 0.0.0.0:8080
```
Comment (AnomalRoil) [inline docker/README.md:38]: same

File: docker/README.md [hunk 4]
```diff
@@ -1,6 +1,70 @@
...
+2023-02-08T14:08:42.643Z	INFO	0.0.0.0:8080	core/drand_daemon.go:145	DrandDaemon initialized	{"private_listen": "0.0.0.0:8080", "control_port": "8888", "public_listen": "", "folder": "/data/drand/.drand/multibeacon"}
+2023-02-08T14:08:42.679Z	INFO	0.0.0.0:8080	core/drand_daemon.go:316	beacon id [default]: will run as fresh install -> expect to run DKG.
+```
+
+If we try and run `curl -v localhost:8080/chains` we won't get anything back! The private listening port only speaks gRPC and is used when
+nodes talk to one another. To expose the randomness itself, we must provide a public listening port.
+
+Kill the container and rerun it with a command such as:
+```shell
+docker run --rm -d -p"8080:8080" -p"8888:8888" -p"9080:9080" --name drand  --volume drand:/data/drand drandorg/go-drand:v1.5.3 start --tls-disable --private-listen 0.0.0.0:8080 --public-listen 0.0.0.0:9080
```
Comment (AnomalRoil) [inline docker/README.md:54]: same

File: docker/docker-compose-network.yml
Comment (AnomalRoil) [inline docker/docker-compose-network.yml:1]: Same for the images here

File: docker/docker-compose-nginx.yml
Comment (AnomalRoil) [inline docker/docker-compose-nginx.yml:1]: and same

File: docker/docker-compose.yml
Comment (AnomalRoil) [inline docker/docker-compose.yml:1]: you guessed it

File: docker/start-network-with-nginx.sh
Comment (AnomalRoil) [inline docker/start-network-with-nginx.sh:1]: same

File: docker/start-network.sh
Comment (AnomalRoil) [inline docker/start-network.sh:1]: same

**Review by AnomalRoil — APPROVED** [review]
Let me know if you'd prefer to do it in another PR, happy to get this merged already.

File: docker/README.md (hunk 1 above)
Comment (CluEleSsUK) [reply docker/README.md:12]: Will raise another PR once we get a tagged version pushed

**Pull request merged**